### ISSUE TITLE
[xpu-aglinment] Fix channels-last 1x1 convolution backward gradients

### ISSUE THREAD

**Issue by laifenxiawucha** [intel/torch-xpu-ops#4756]
### 🐛 Describe the bug

XPU oneDNN convolution backward can silently produce wrong gradients for
channels-last 1x1 spatial or kernel shapes. The backward code derives its
`is_channels_last` flag from a tensor pair that can be layout-nondiscriminating
at these shapes, even when the caller-allocated `diff_weight` or `diff_src`
buffer is channels-last. oneDNN then describes that buffer with the wrong layout
and reads or writes it using incorrect strides.

This was found during XPU alignment for upstream PyTorch PR
https://github.com/pytorch/pytorch/pull/190773, which already contains the direct
XPU backend fix. The affected integration lives in the PyTorch tree
(`aten/src/ATen/native/mkldnn/xpu/detail/Conv.cpp`); this issue is the XPU
evidence tracker for that correction.

### Reproducer

```python
import torch

torch.manual_seed(42)
input_ref = torch.randn((8, 16, 1, 1), device="xpu", requires_grad=True)
weight_ref = torch.randn((16, 16, 3, 3), device="xpu", requires_grad=True)
grad = torch.randn(
    torch.conv2d(input_ref, weight_ref, padding=1).shape, device="xpu"
)

# Contiguous reference.
torch.conv2d(input_ref, weight_ref, padding=1).backward(grad)
expected_weight_grad = weight_ref.grad.clone()

# Equivalent channels-last weight path.
input_cl = input_ref.detach().requires_grad_()
weight_cl = weight_ref.detach().to(memory_format=torch.channels_last).requires_grad_()
torch.conv2d(input_cl, weight_cl, padding=1).backward(grad)

assert weight_cl.grad.device.type == "xpu"
weight_grad_diff = (weight_cl.grad - expected_weight_grad).abs().max().item()
print(f"weight_grad_diff={weight_grad_diff:.2e}")
assert weight_grad_diff < 1e-4
```

Also cover the backward-data case with a channels-last input, shape
`(8, 16, 8, 8)`, a `1x1` `(32, 16, 1, 1)` weight, and stride `(2, 2)`.

### Actual behavior

```text
torch version: 2.14.0.dev20260713+xpu
xpu available: True
Case 1 (1x1-spatial, CL weight): input_grad_diff=0.00e+00, weight_grad_diff=8.34e+00
Case 2 (1x1 kernel, CL input): input_grad_diff=3.81e-06, weight_grad_diff=5.72e-06
Case 1 gradient mismatch (1x1 spatial, channels-last weight)
RESULT: confirmed
```

The `8.34e+00` weight-gradient divergence is a silent XPU numerical miscompute,
not a test, shape, or dispatch-presence failure.

### Expected behavior

Channels-last convolution backward gradients match the contiguous reference
(`weight_grad_diff < 1e-4`) for both the 1x1-spatial weight case and the
stride-2 1x1 backward-data case.

### Alignment metadata

- Upstream source: https://github.com/pytorch/pytorch/pull/190773
- Source type: upstream PR (fix lands in PyTorch XPU oneDNN integration)
- Local XPU result: confirmed
- Routed area: `pytorch/pytorch`, `aten/src/ATen/native/mkldnn/xpu/detail/Conv.cpp` (not torch-xpu-ops source)

### Notes

In the XPU oneDNN backward path, `is_channels_last` is derived from tensor pairs
that exclude the caller-allocated output buffer. At 1x1 spatial activations or
1x1 kernels, channels-last and contiguous strides coincide, so
`suggest_memory_format` can report contiguous even though the output buffer uses
channels-last storage. The resulting oneDNN descriptors label `diff_weight` as
`oihw` instead of `ohwi`, or `diff_src` as `nchw` instead of `nhwc`, and oneDNN
returns silently wrong gradients.

The fix derives the layout flag from the layout-discriminating caller-allocated
buffer (`diff_weight` in `convolution_backward_weights`, `diff_src` in
`convolution_backward_data`). A regression `test_conv_backward_channels_last_1x1`
should cover both the 1x1-spatial weight case and the stride-2 1x1 backward-data
case, failing on the pre-fix implementation and passing after a rebuilt XPU
source build.

### Versions

```text
PyTorch version: 2.14.0.dev20260713+xpu
OS: Ubuntu 22.04.5 LTS (x86_64)
Is XPU available: True
Intel GPU: Intel(R) Data Center GPU Max 1100
Relevant path: aten/src/ATen/native/mkldnn/xpu/detail/Conv.cpp
```
